### PR TITLE
add mysql aliases for Text and Binary types

### PR DESCRIPTION
**Pull request by Fiedzia**
Handle mysql text and binary types by adding suitable aliases
We may want to have something more ergonomic for bit type

### CODE REVIEW

File: diesel/src/types/mod.rs [hunk 1]
```diff
@@ -174,6 +174,8 @@ impl SingleValue for Numeric {}
 /// On all backends strings must be valid UTF-8.
 /// On PostgreSQL strings must not include nul bytes.
 ///
+/// On Mysql, it is also aliased by Tinytext, Mediumtext, Longtext, Char and Varchar.
```
Comment (killercup) [inline diesel/src/types/mod.rs]: Just cosmetics, but can you change it to "MySQL" and put the type names in back ticks (like <code>\`this\`</code>).

File: diesel/src/types/mod.rs [hunk 2]
```diff
@@ -188,9 +190,15 @@ impl SingleValue for Numeric {}
...
 pub type VarChar = Text;
 #[doc(hidden)] pub type Varchar = VarChar;
+#[doc(hidden)] pub type Char       = Text;
+#[doc(hidden)] pub type Tinytext   = Text;
+#[doc(hidden)] pub type Mediumtext = Text;
+#[doc(hidden)] pub type Longtext   = Text;
 
 /// The binary SQL type.
 ///
+/// On Mysql, it is also aliased by Tinyblob, Mediumblob, Longblob, Bit and Varbinary.
```
Comment (killercup) [inline diesel/src/types/mod.rs]: Same as above

**Comment by Fiedzia** [comment]
> I'm not entirely sure what the semantic differences of these types are

None I believe. If I read MySQL docs correctly, Mysql will handle them differently on the server side (at storage level), but there is no difference in the transport layer. At storage level those types simply differ in allowed max size.

> What use cases does this add for you?

Schema generated by diesel print-schema compiles. Also I'd like to have ability to generate create table sql from diesel schema definition (I am not sure if/how that works yet, but being specific here would help in achieving that).

File: diesel/src/types/mod.rs [hunk 3]
```diff
@@ -188,9 +190,15 @@ impl SingleValue for Numeric {}
 #[derive(Debug, Clone, Copy, Default)] pub struct Text;
 pub type VarChar = Text;
 #[doc(hidden)] pub type Varchar = VarChar;
+#[doc(hidden)] pub type Char       = Text;
```
Comment (sgrif) [inline diesel/src/types/mod.rs]: Please avoid visual alignment.

**Comment by Fiedzia** [comment]
This is part of solution for #830

**Comment by killercup** [comment]
@sgrif are you good with merging this? I'd probably want to add a test for inferring all these types, but I think that's not that easy right now.

**Comment by sgrif** [comment]
LGTM

**Comment by killercup** [comment]
Thanks again!

**Pull request merged**